src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java

```diff
@@ -0,0 +1,98 @@
+        lfParametersExt.setThrowsExceptionInCaseOfSlackDistributionFailure(false);
+
+        // load contingencies
+        List<Contingency> contingencies = contingenciesProvider.getContingencies(network);
+
+        // try to find all switches impacted by at least one contingency and for each contingency the branches impacted
+        Set<Switch> allSwitchesToOpen = new HashSet<>();
+        sensiDC = new DcSensitivityAnalysis(matrixFactory, connectivityProvider);
+
+        OpenSensitivityAnalysisProvider sensitivityAnalysisProvider = new OpenSensitivityAnalysisProvider();
+
+        List<SensitivityVariableSet> variableSets = new ArrayList<>();
+        SensitivityAnalysisParameters sensitivityAnalysisParameters = new SensitivityAnalysisParameters();
+        sensitivityAnalysisParameters.getLoadFlowParameters().setDc(true);
+
+        ContingencyContext contingencyContext = new ContingencyContext(null, ContingencyContextType.ALL);
+        String variableId = network.getLoads().iterator().next().getId();
+
+        List<SensitivityFactor2> factors = new ArrayList<>();
+        for (Branch b : network.getBranches()) {
+            factors.add(new SensitivityFactor2(SensitivityFunctionType.BRANCH_ACTIVE_POWER, b.getId(), SensitivityVariableType.INJECTION_ACTIVE_POWER,
+                    variableId, false, contingencyContext));
+        }
+        SensitivityAnalysisResult2 res = sensitivityAnalysisProvider.run(network, contingencies, variableSets, sensitivityAnalysisParameters, factors);
+
+        DefaultLimitViolationDetector detector = new DefaultLimitViolationDetector(1.0f, EnumSet.allOf(LoadingLimitType.class));
+
+        List<LimitViolation> preContingencyLimitViolations = new ArrayList<>();
+        for (SensitivityValue2 sensValue : res.getValues(null)) {
+            SensitivityFactor2 factor = (SensitivityFactor2) sensValue.getFactorContext();
+            String branchId = factor.getFunctionId();
+            Branch branch = network.getBranch(branchId);
+            detector.checkActivePower(branch, Branch.Side.ONE, Math.abs(sensValue.getFunctionReference()), preContingencyLimitViolations::add);
+        }
+
+        LimitViolationsResult preContingencyResult = new LimitViolationsResult(true, preContingencyLimitViolations);
+
+        List<PostContingencyResult> postContingencyResults = new ArrayList<>();
+        for (Contingency contingency : contingencies) {
+            List<SensitivityValue2> values = res.getValues(contingency.getId());
+            List<LimitViolation> violations = new ArrayList<>();
+
+            for (SensitivityValue2 v : values) {
+                SensitivityFactor2 factor = (SensitivityFactor2) v.getFactorContext();
+                String branchId = factor.getFunctionId();
+                Branch branch = network.getBranch(branchId);
+                detector.checkActivePower(branch, Branch.Side.ONE, Math.abs(v.getFunctionReference()), violations::add);
```

Comment:
Just a question: in the detector, do we have the rule as we have implemented here with `POST_CONTINGENCY_INCREASING_FACTOR`? If not, maybe we have to think at the implementation on core side.

Reply:
POST_CONTINGENCY_INCREASING_FACTOR seems to be used as a post process to modify postContingencyLimitViolations list in relation with pre contingency list. As far as I know, there is nothing equivalent implemented yet in DC mode.